Bottom: 8c77753f2d2bb7b437274dc50d04cab82101c168
Top:    ed5f8b2d65a70d1b508c400e0b98dae46dcfcca3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-14 18:52:41 +0000

Refresh of improve-grid-snapping-heuristi

---

diff --git a/src/search.c b/src/search.c
index 153f60c..5e86cc1 100644
--- a/src/search.c
+++ b/src/search.c
@@ -320,8 +320,7 @@ SearchRatLineByLocation (int locked, RatTypePtr * Line, RatTypePtr * Dummy1,
  */
 struct arc_info
 {
-  ArcTypePtr *Arc, *Dummy;
-  jmp_buf env;
+  ArcTypePtr *Arc;
   int locked;
   int smallest_radius;
 };
@@ -345,10 +344,8 @@ arc_callback (const BoxType * box, void *cl)
     {
       i->smallest_radius = found_radius;
       *i->Arc = a;
-      *i->Dummy = a;
     }
-//  longjmp (i->env, 1);
-  return 1;//			/* never reached */
+  return 1;
 }
 
 
@@ -359,16 +356,11 @@ SearchArcByLocation (int locked, LayerTypePtr * Layer, ArcTypePtr * Arc,
   struct arc_info info;
 
   info.Arc = Arc;
-  info.Dummy = Dummy;
   info.locked = (locked & LOCKED_TYPE) ? 0 : LOCKFLAG;
   info.smallest_radius = -1;
 
   *Layer = SearchLayer;
-//  if (setjmp (info.env) == 0)
-//    {
-      r_search (SearchLayer->arc_tree, &SearchBox, NULL, arc_callback, &info);
-//      return False;
-//    }
+    r_search (SearchLayer->arc_tree, &SearchBox, NULL, arc_callback, &info);
   if (info.smallest_radius > -1)
     return True;
   else
